kt/godot-library/src/main/kotlin/godot/core/memory/GodotBinding.kt

```diff
@@ -0,0 +1,41 @@
+package godot.core.memory
+
+import godot.core.KtObject
+import godot.core.ObjectID
+import java.lang.ref.ReferenceQueue
+import java.lang.ref.WeakReference
+
+internal interface GodotBinding {
+    val objectID: ObjectID
+    val instance: KtObject?
+
+    companion object {
+        /** Queue to be notified when the GC runs on References.*/
+        val queue = ReferenceQueue<KtObject>()
+
+        fun create(instance: KtObject): GodotBinding {
+            val id = instance.objectID
+            return if (id.isReference) {
+                RefCountedBinding(instance, queue, id)
+            } else {
+                ObjectBinding(instance, id)
+            }
+        }
+    }
+}
+
+internal class RefCountedBinding(
+    instance: KtObject,
+    queue: ReferenceQueue<KtObject>,
+    override val objectID: ObjectID
+) : WeakReference<KtObject>(instance, queue), GodotBinding {
+
+    override val instance
+        get() = get()
```

Comment:
Please explicitly write the type and use `get() = thi.get()`
this hurts my eyes just looking at `RefCountedBinding` alone ^^

```suggestion
    override val instance: KtObject?
        get() = this.get()
```